xls/public/c_api.cc

```diff
@@ -428,6 +455,50 @@ bool xls_bits_get_bit(const struct xls_bits* bits, int64_t index) {
   return cpp_bits->Get(index);
 }
 
+bool xls_bits_to_bytes(const struct xls_bits* bits, char** error_out,
+                       uint8_t** bytes_out, size_t* byte_count_out) {
+  CHECK(bits != nullptr);
+  CHECK(bytes_out != nullptr);
+  CHECK(error_out != nullptr);
+  CHECK(byte_count_out != nullptr);
+  const auto* cpp_bits = reinterpret_cast<const xls::Bits*>(bits);
+  auto bytes = cpp_bits->ToBytes();
+  *byte_count_out = bytes.size();
+  *bytes_out = (uint8_t*)malloc(bytes.size());
```

Comment:
You'll need a `xls_bytes_free()` for this (because the libxls allocator can be different from the surrounding process so you can't assume the surrounding process can use the same `free()`), and generally I think we've been doing new[] inside the impl e.g. https://sourcegraph.com/github.com/google/xls@ef07f4bc7db8df70dc32e52a04d268653b58847a/-/blob/xls/public/c_api_impl_helpers.cc?L72

Reply:
Done, can you take another look?